scipy/stats/tests/test_continuous_basic.py

```diff
@@ -186,58 +190,79 @@ def test_cont_basic_slow():
                 arg = (1,)
             elif distname == 'ksone':
                 arg = (3,)
-            yield check_named_args, distfn, x, arg, locscale_defaults, meths
-            yield check_random_state_property, distfn, arg
-            yield check_pickling, distfn, arg
+            check_named_args(distfn, x, arg, locscale_defaults, meths)
+            check_random_state_property(distfn, arg)
+            check_pickling(distfn, arg)
 
             # Entropy
-            skp = npt.dec.skipif
-            ks_cond = distname in ['ksone', 'kstwobign']
-            yield skp(ks_cond)(check_entropy), distfn, arg, distname
+            if distname not in ['ksone', 'kstwobign']:
+                check_entropy(distfn, arg, distname)
 
             if distfn.numargs == 0:
-                yield check_vecentropy, distfn, arg
+                check_vecentropy(distfn, arg)
             if (distfn.__class__._entropy != stats.rv_continuous._entropy
                     and distname != 'vonmises'):
-                yield check_private_entropy, distfn, arg, stats.rv_continuous
+                check_private_entropy(distfn, arg, stats.rv_continuous)
+
+            check_edge_support(distfn, arg)
+
+            check_meth_dtype(distfn, arg, meths)
+            check_ppf_dtype(distfn, arg)
 
-            yield check_edge_support, distfn, arg
+            if distname not in fails_cmplx:
+                check_cmplx_deriv(distfn, arg)
 
-            yield check_meth_dtype, distfn, arg, meths
-            yield check_ppf_dtype, distfn, arg
-            yield skp(distname in fails_cmplx)(check_cmplx_deriv), distfn, arg
+    for distname, arg in distcont[:]:
+        if distname not in distslow:
+            continue
+
+        if distname is 'levy_stable':
+            continue
+
+        yield check, distname, arg
 
 
 @npt.dec.slow
 def test_moments():
-    with warnings.catch_warnings():
-        warnings.filterwarnings('ignore',
-                                category=integrate.IntegrationWarning)
-        knf = npt.dec.knownfailureif
-        fail_normalization = set(['vonmises', 'ksone'])
-        fail_higher = set(['vonmises', 'ksone', 'ncf'])
-        for distname, arg in distcont[:] + [(test_histogram_instance, tuple())]:
+    def check(distname, arg, normalization_ok, higher_ok):
```

Comment:
Didn't do it as it makes it more difficult to review...
If we'll go pytest, there's even more rewriting to do here so I'd leave it for later.